server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -290,6 +294,9 @@ protected void doRun() {
     public void applyClusterState(ClusterChangedEvent event) {
         try {
             final ClusterState state = event.state();
+            if (isDedicatedVotingOnlyNode(state.nodes().getLocalNode())) {
```

Comment:
Maybe we could just not add the applier in the first place in the constructor where we check if the node is a master or data node (we should have the information that this is a voting only node there already from the settings?)?

Reply:
We need to be able to compute this from a `DiscoveryNode` since we need to know it for remote nodes too, and we won't have a `DiscoveryNode` in the constructor so would have to implement it two different ways. I think I prefer this.

Reply:
yeah, I went back and forth on this, and prefer to reuse the same logic.

I have done this in a different way now in 143fc1cf83e: Let me know if you prefer this.

Reply:
I like it :)